Essentials/src/main/java/com/earth2me/essentials/Kits.java

```diff
@@ -31,23 +33,49 @@ public void reloadConfig() {
         kits = _getKits();
     }
 
+    private void addKitSectionToNode(final String kitName, final CommentedConfigurationNode kitSection, final CommentedConfigurationNode destination) {
+        if (kitSection.isMap()) {
+            try {
+                destination.node(kitName.toLowerCase(Locale.ENGLISH)).set(kitSection);
+            } catch (SerializationException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
     private CommentedConfigurationNode _getKits() {
-        final CommentedConfigurationNode section = config.getSection("kits");
-        if (section != null) {
-            final CommentedConfigurationNode newSection = config.newSection();
-            for (final String kitItem : ConfigurateUtil.getKeys(section)) {
-                final CommentedConfigurationNode kitSection = section.node(kitItem);
-                if (kitSection.isMap()) {
-                    try {
-                        newSection.node(kitItem.toLowerCase(Locale.ENGLISH)).set(kitSection);
-                    } catch (SerializationException e) {
-                        e.printStackTrace();
+        final CommentedConfigurationNode newSection = config.newSection();
+
+        // Kits from kits.yml file
+        final CommentedConfigurationNode fileKits = config.getSection("kits");
+        if (fileKits != null) {
+            for (final String kitItem : ConfigurateUtil.getKeys(fileKits)) {
+                addKitSectionToNode(kitItem, fileKits.node(kitItem), newSection);
+            }
+        }
+
+        // Kits from kits subdirectory
+        final File kitsFolder = new File(this.ess.getDataFolder(), "kits");
+        if (!kitsFolder.exists() || !kitsFolder.isDirectory()) {
+            return newSection;
+        }
+
+        final File[] kitsFiles = kitsFolder.listFiles();
+
+        //noinspection ConstantConditions - will not be null, conditions checked above.
+        for (final File kitFile : kitsFiles) {
+            if (kitFile.getName().endsWith(".yml")) {
+                final EssentialsConfiguration kitConfig = new EssentialsConfiguration(kitFile);
+                kitConfig.load();
+                final CommentedConfigurationNode kits = kitConfig.getSection("kits");
+                if (kits != null) {
+                    for (final String kitItem : ConfigurateUtil.getKeys(kits)) {
+                        addKitSectionToNode(kitItem, kits.node(kitItem), newSection);
                     }
                 }
             }
-            return newSection;
         }
-        return null;
+        return newSection;
```

Comment:
The `matchKit` and `listKits` methods ignore the `kits` field and load directly from the config. We probably want these methods to use `kits` instead.

In addition, do we want `addKit` and `removeKit` to work on `kits/` files? If so, we'd need to change the syntax of `/createkit` and `/delkit` to accommodate this, and probably add overloads to the existing methods.

Personally I'm okay with only allowing `/createkit` and `/delkit` to touch the `kits.yml` (if a server admin needed was managing kits using multiple files, I'd assume it's because they prefer to modify the files directly). However, there's not much stopping us from allowing this if we wanted to.